Bug: Script fails for records without ARKs

Now that we have a mix of records with and without ARKs, it's now breaks in cases where one does not exist (new records):

```python
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.7/x64/bin/scraper", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/runner/work/pdc-osti/pdc-osti/src/pdc_osti/scraper.py", line 421, in main
    s.run_pipeline()
  File "/home/runner/work/pdc-osti/pdc-osti/src/pdc_osti/scraper.py", line 361, in run_pipeline
    self.get_unposted_metadata()
  File "/home/runner/work/pdc-osti/pdc-osti/src/pdc_osti/scraper.py", line 2[28](https://github.com/pulibrary/pdc-osti/actions/runs/7376863887/job/20070155428#step:6:29), in get_unposted_metadata
    princeton_handles = [get_ark(record) for record in princeton_j]
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/pdc-osti/pdc-osti/src/pdc_osti/scraper.py", line 228, in <listcomp>
    princeton_handles = [get_ark(record) for record in princeton_j]
                         ^^^^^^^^^^^^^^^
  File "/home/runner/work/pdc-osti/pdc-osti/src/pdc_osti/commons.py", line 4, in get_ark
    return record["resource"]["ark"].replace("ark:/", "")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'replace'
Error: Process completed with exit code 1.
```

A try and exception should be pursued.

--- a/src/pdc_osti/commons.py
+++ b/src/pdc_osti/commons.py
@@ -1,7 +1,10 @@
 def get_ark(record: dict) -> str:
     """Retrieves ARK (e.g., 88435/dsp012j62s808w) depending on Princeton source"""
 
-    return record["resource"]["ark"].replace("ark:/", "")
+    try:
+        return record["resource"]["ark"].replace("ark:/", "")
+    except AttributeError:
+        return ""
 
 
 def get_author(creator: dict) -> dict:

--- a/src/pdc_osti/scraper.py
+++ b/src/pdc_osti/scraper.py
@@ -225,7 +225,7 @@ class Scraper:
             json.dump(to_be_published, f, indent=4)
 
         # Check for records in OSTI but not PDC
-        princeton_handles = [get_ark(record) for record in princeton_j]
+        princeton_handles = [get_doi(record) for record in princeton_j]
         errors = [
             record
             for record in osti_j

--- a/src/pdc_osti/scraper.py
+++ b/src/pdc_osti/scraper.py
@@ -229,7 +229,7 @@ class Scraper:
         errors = [
             record
             for record in osti_j
-            if redirects_j[record["doi"]] not in princeton_handles
+            if record["doi"].replace("https://doi.org/", "") not in princeton_handles
         ]
         if len(errors) > 0:
             self.log.warning(

